helm/templates/statefulset.yaml

```
@@ -0,0 +1,74 @@
+apiVersion: apps/v1
+kind: StatefulSet
+metadata:
+  name: {{ include "immudb.fullname" . }}
+  labels:
+    {{- include "immudb.labels" . | nindent 4 }}
+spec:
+  replicas: {{ .Values.replicaCount }}
+  selector:
+    matchLabels:
+      {{- include "immudb.selectorLabels" . | nindent 6 }}
+  serviceName: {{ include "immudb.fullname" . }}    
+  template:
+    metadata:
+      {{- with .Values.podAnnotations }}
+      annotations:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+      labels:
+        {{- include "immudb.selectorLabels" . | nindent 8 }}
+    spec:
+      {{- with .Values.imagePullSecrets }}
+      imagePullSecrets:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+      serviceAccountName: {{ include "immudb.serviceAccountName" . }}
+      securityContext:
+        {{- toYaml .Values.podSecurityContext | nindent 8 }}
+      containers:
+        - name: {{ .Chart.Name }}
+          securityContext:
+            {{- toYaml .Values.securityContext | nindent 12 }}
+          image: "{{ .Values.image.repository }}:{{ .Values.image.tag | default .Chart.AppVersion }}"
+          imagePullPolicy: {{ .Values.image.pullPolicy }}
+          ports:
+            - name: http
+              containerPort: 8080
+              protocol: TCP
+            - name: metrics
+              containerPort: 9497
+              protocol: TCP
+            - name: grpc
+              containerPort: 3322
+              protocol: TCP
+          livenessProbe:
+            httpGet:
+              path: /readyz
+              port: metrics
+          readinessProbe:
+            httpGet:
+              path: /readyz
+              port: metrics
+          {{- if $.Values.adminPassword }}
```

It would be usefull to also specify existing secret that contains the password, like:
```suggestion
          {{- if $.Values.adminPasswordExistingSecret }}
          .....
          {{- else if $.Values.adminPassword }}
```

Another thing that I see more of a future update

---

It can be useful, but I'll focus on the others point for the moment.